### ISSUE TITLE
`Execute()` is not sending request to the correct URL

### ISSUE THREAD

**Issue by marcustut** [supabase/postgrest-go#3]
# Bug report

## Describe the bug

So when I try to insert data with the following code snippet, I expect it to send a request to `https://<supabase_url>/rest/v1/products` but it seems like the `Execute()` function is not sending the request to the correct URL.
![image](https://user-images.githubusercontent.com/59773847/120691924-aaf40c00-c4d9-11eb-99d4-d603ea7df8e2.png)

The result of trying to run this is that I get the response as below, after a google search it seems like this is what Kong would respond if it can't find a route

![image](https://user-images.githubusercontent.com/59773847/120692283-2950ae00-c4da-11eb-95c4-9f766f880329.png)

So I try to directly change the `client.clientTransport.baseURL.Path` to `https://<supabase_url>/rest/v1/products`, the request is sent to the correct place where I can see a postgrest response but the query is still not successfully created yet. I'm suspecting is that the way the querybuilder marshal json is making some errors here.

![image](https://user-images.githubusercontent.com/59773847/120691889-9e6fb380-c4d9-11eb-926a-3d329a5a26fb.png)

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Get the library
2. Use a supabase url, anon key and proper auth token to create the client
3. Try to insert data into a table

## Expected behavior

The request should be sent to the correct url and querybuilder should properly marshal `value` into json

## Screenshots

Shown above

## System information

- OS: macOS Big Sur 11.4
- Version of Go: 1.16

## Additional context

I will try fixing it and if I do, I will create a PR.

**Comment by marcustut** [comment]
Looking further into the URL issue, it seems that because local postgrest would run on an url like `http://localhost:3000` and therefore the path to a table would be `http://localhost:3000/<table_name>` but since supabase has its postgrest server at `https://<supabase_url>/rest/v1` so the path to a table would be `https://<supabase_url>/rest/v1/<table_name>`.

But if I directly pass `https://<supabase_url>/rest/v1` to the NewClient func, it wouldn't work as well due to how url parsing is implemented, it will result in the following baseURL

![image](https://user-images.githubusercontent.com/59773847/120695654-38395f80-c4de-11eb-954c-18c78ae1b1ac.png)

```go
// So doing the following would result in a correct path

client.From("/rest/v1/products")

// instead of

client.From("products")
```

probably we should set a func param while creating the client to specify the nested path?

**Comment by yusufpapurcu** [comment]
No we must be able to give url like first example. I will look at this issue.

**Comment by yusufpapurcu** [comment]
Can you try adding `https://` in your url?

**Comment by marcustut** [comment]
> Can you try adding `https://` in your url?

It's already added, this is how I specify it

![image](https://user-images.githubusercontent.com/59773847/120697465-556f2d80-c4e0-11eb-9eef-2c431fee9831.png)

**Comment by yusufpapurcu** [comment]
<img width="691" alt="Screen Shot 2021-06-03 at 21 58 45" src="https://user-images.githubusercontent.com/41443083/120697880-f1843f00-c4b6-11eb-929a-570ed89434b1.png">

You must define like this. This is just a postgrest library and you must specify postgrest path. Supabase's postgrest runs in `rest/v1/`. That's cause of your problem 😆

**Comment by marcustut** [comment]
> <img alt="Screen Shot 2021-06-03 at 21 58 45" width="691" src="https://user-images.githubusercontent.com/41443083/120697880-f1843f00-c4b6-11eb-929a-570ed89434b1.png">
> 
> You must define like this. This is just a postgrest library and you must specify postgrest path. Supabase's postgrest runs in `rest/v1/`. That's cause of your problem 😆

Ah.. yea tested it just now, my bad. turns out you need to set `https://<supabase_url>/rest/v1/` for it to work instead of `https://<supabase_url>/rest/v1`

However, I still can't get `Insert()` to work, still trying to debug it, however `Select()` works flawlessly.

**Comment by marcustut** [comment]
I got `insert()` to work, the problem is with me again 😅  I was passing `map[string]interface{}` as `value` which `json.Marshal` doesn't know how to marshal it into a correct json object.

So I wrote a struct with proper json struct tags that correspond to the table's field name then it worked. 

Anyways, I found that the examples is quite lacking, shall I write few more examples under the `test/` folder?
